src/test/java/StringManipulationTest.java

```diff
@@ -26,21 +26,49 @@ public void testCount1() {
         assertEquals(4, length);
     }
 
+    // Checks what happens if an empty string is passed in
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("          ");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // This test checks that numbers are recognized as strings
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("132 444");
+        int length = manipulatedstring.count();
+        assertEquals(2, length);
     }
 
+    // This test ensures that the counter works on a long string
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        StringBuilder stringBuilder = new StringBuilder();
+        String phrase = "the quick brown fox jumps over the lazy dog";
+        int occurrences = 0;
+    
+        for (int i = 0; i < 100; i++) {
+            stringBuilder.append(phrase);
+            stringBuilder.append("\n");
+        }
+    
+        String longString = stringBuilder.toString();
+        int index = 0;
+        while (index < longString.length()) {
+            index = longString.indexOf(phrase, index);
+            if (index == -1) {
+                break;
+            }
+            occurrences++;
+            index += phrase.length();
+        }
+    
+        assertEquals(100, occurrences);
```

Comment:
testCount4( ):
I'm sure this is a working code, but I don't think you actually used count() in it...
You might want to rewrite this test case so that it tests the count() in StringManipulation class. 
You could write a test case for a long string or even for an empty string.